cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler.kt

```diff
@@ -537,6 +527,35 @@ class StatementHandler(frontend: PythonLanguageFrontend) :
         return wrapDeclarationToStatement(result)
     }
 
+    private fun handleGlobal(global: Python.AST.Global): LookupScopeStatement {
+        // Technically, our global scope is not identical to the python "global" scope. The reason
+        // behind that is that we wrap each file in a namespace (as defined in the python spec). So
+        // the "global" scope is actually our current namespace scope.
+        var pythonGlobalScope =
+            frontend.scopeManager.globalScope?.children?.firstOrNull { it is NameScope }
+
+        return newLookupScopeStatement(
+            global.names.map { parseName(it).localName },
+            pythonGlobalScope,
+            rawNode = global
+        )
+    }
+
+    private fun handleNonLocal(global: Python.AST.Nonlocal): LookupScopeStatement {
+        // We need to find the first outer function scope, or rather the block scope belonging to
+        // the function
+        var outerFunctionScope =
+            frontend.scopeManager.firstScopeOrNull {
+                it is BlockScope && it != frontend.scopeManager.currentScope
```

Comment:
Do we need to consider nested functions here, too?

Reply:
Unfortunately, I cannot test this, as we are not parsing nested function at all at the moment (see #1736), but I am beginning to hate python even more. It seems that the nonlocal statement actually jumps to the nested function scope where the variable is "declared" and not just the next one. See example
```
def level1():
    c = 1
    def level2():
        d = 1
        def level3():
            nonlocal c, d
            c = c + 2
            d = d + 1
            print(c)
        
        level3()
    level2()
level1()
```

In this case the nonlocal c "jumps" to level 1 and nonlocal d jumps to level 2, but we cannot retrieve this information in the frontend so I would take this as a close approximation and ignore this very narrow corner case.

Reply:
Will be addressed in #1736 -> resolved for now.